cpp/src/cider/benchmarks/HashMapBenchmark.cpp

```diff
@@ -114,6 +116,59 @@ static std::tuple<std::vector<T>, std::vector<bool>> generateAndFillVector(
 template <typename KeyType, typename HashTableType>
 static void BM_Lookup(benchmark::State& state) {
   size_t row_num = state.range(0);
+  std::vector<KeyType> data = prepare_data<KeyType>(row_num);
+  for (auto _ : state) {
+    int32_t collisions = bench<KeyType, HashTableType>(data);
+    state.counters["Collisions"] = collisions;
+  }
+}
+
+template <typename KeyType>
+void NO_INLINE bench_std(const std::vector<KeyType>& data) {
+  std::map<KeyType, int8_t> map;

```

Comment:
please use `std::unordered_map` instead.

Reply:
Keep `std::map` in this pr and add `std::unordered_map` later.

Reply:
`std::map` is not a hash map implementation. feel free to keep it here if you think it's necessary.

Reply:
It's necessary since `std::map` is something like `TreeMap` in java, which doesn't use hash function but will be more efficient than `std::unordered_map` in many cases.

Reply:
It seems this function does nothing for input data
for benchmark, you should use some info of map to avoid this function be optimized nothing executed under O3
such like ``` return map.size();```

Reply:
maybe google benchmark will stop compiler optimize this function.